src/lib/utils/assert.h

```diff
@@ -112,10 +112,10 @@ namespace Botan {
 * BOTAN_UNUSED(a); or BOTAN_UNUSED(x, y, z);
 */
 template <typename T>
-constexpr void ignore_param(T&&) {}
+constexpr void ignore_param(const T&) {}
 
 template <typename... T>
-constexpr void ignore_params(T&&... args) {
+constexpr void ignore_params(const T&... args) {
```

Comment:
Unrelated to clang-tidy, but [the non-variadic overload isn't necessary](https://godbolt.org/z/PEbeWf947) here. Like so:

```C+
template <typename... Ts>
constexpr void ignore_params(const Ts&...) {}
```